sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuWindowExpression.scala

```diff
@@ -166,6 +173,11 @@ class GpuWindowExpressionMeta(
       childExprs.head.convertToGpu(),
       childExprs(1).convertToGpu().asInstanceOf[GpuWindowSpecDefinition]
     )
+
+  // Allow array type only for Python UDF, also add an extra check in
+  // `tagExprForGpu` to make sure this.
+  override def isSupportedType(t: DataType): Boolean =
+    GpuOverrides.isSupportedType(t, allowArray = true, allowNesting = true)
```

Comment:
We have not tested nested arrays in the output i.e.  `ArrayType(ArrayType(LongType))` I would prefer that we don't explicitly allow it without a test for it.

Reply:
Nice catch.
I meant one level array. Updated. Looks like i did not fully understand this API `GpuOverrides.isSupportedType`.